File: CTrove.Core/Entity/Organization.cs
using CTrove.Core.Enum;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CTrove.Core.Entity
{
    public class Organization
    {
        [Key]
        public Guid Id { get; set; }
        public Guid CountryId { get; set; }
       // public virtual Country Country { get; set; }
        public string CompanyName { get; set; } = string.Empty;
        public ParentType Parent { get; set; }
        //public string Parent { get; set; } = string.Empty;
        public Guid ContactTypeId { get; set; }
        public virtual ContactType ContactType { get; set; }
        public Guid VendorTypeId { get; set; }
        public virtual VendorType VendorType { get; set; }
        public string? AddressLine1 { get; set; } = string.Empty;
        public string? AddressLine2 { get; set; } = string.Empty;
        public string? AddressLine3 { get; set; } = string.Empty;
        public string? ZipCode { get; set; } = string.Empty;
        public string? City { get; set; } = string.Empty;
        public string? State { get; set; } = string.Empty;
        public string? PhoneNumber { get; set; } = string.Empty;
        public string? FaxNumber { get; set; } = string.Empty;
        public string? Website { get; set; } = string.Empty;
        public Guid? PrimaryContactId { get; set; }
      //  public virtual Contributor PrimaryContributorContact { get; set; }
        public Guid? SecondaryContactId { get; set; }
       // public virtual Contributor SecondaryContributorContact { get; set; }
        public string? Email { get; set; } = string.Empty;
        public string? Notes { get; set; } = string.Empty;
        public bool Status { get; set; }
    }
}

File: CTrove/angularapp/src/app/models/contributor/organization.ts
import { ADD, SEARCH, STATUS } from 'src/Utilities/common/app-strings';
import { Base, BaseFilter, Deserializable } from '../dto/common';
import { ContactType } from './contact-type';
import { Contributor } from './contributor';
import { VendorType } from './vendor-type';
import { Country } from './country';

export class Organization extends Base implements Deserializable {
  countryId!: string;
  country!: Country;
  companyName!: string;
  parent!: number;
  contactTypeId!: string;
  contactType!: ContactType;
  vendorTypeId!: string;
  vendorType!: VendorType;
  addressLine1!: string;
  addressLine2!: string;
  addressLine3!: string;
  zipCode!: string;
  city!: string;
  state!: string;
  phoneNumber!: string;
  faxNumber!: string;
  website!: string;
  primaryContactId!: string;
  primaryContributorContact!: Contributor;
  secondaryContactId!: string;
  secondaryContributorContact!: Contributor;
  email!: string;
  notes!: string;

  deserialize(input: any): this {
    Object.assign(this, input);
    return this;
  }
}

export class OrganizationFilter extends BaseFilter implements Deserializable {
  countrysId!: string[];
  parentTypes!: number[];
  deserialize(input: any): this {
    Object.assign(this, input);
    return this;
  }
}

// Fields
export const FIELDS = {
  search: SEARCH,
  name: 'Company Name',
  parent: 'Parent',
  contactType: 'Contact Type',
  vendorType: 'Vendor Type',
  country: 'Country',
  status: STATUS,
  primaryContact: 'Primary Contact',
  secondaryContact: 'Secondary Contact',
  email: 'Email',
  website: 'Website',
  state: 'State',
  city: 'City',
  zipCode: 'Zip Code',
  address1: 'Address 1',
  address2: 'Address 2',
  address3: 'Address 3',
  notes: 'Notes',
  phoneNumber: 'Phone Number',
  faxNumber: 'Fax Number',
  address: 'Address',
  contact: 'Contact',
  add: ADD
}
